js/packages/openinference-instrumentation-bedrock-agent-runtime/src/attributes/attributeExtractionUtils.ts

```diff
@@ -883,6 +898,128 @@ export function getFailureTraceAttributes(
+ * Extract attributes from guardrail trace data.
+ * @param guardrailTrace Guardrail trace data object.
+ * @returns Guardrail trace attributes.
+ */
+export function getGuardrailTraceMetadata(
+  guardrailTrace: StringKeyedObject,
+): StringKeyedObject {
+  const guardrailTraceData: StringKeyedObject = {};
+
+  const action = getStringAttributeValueFromUnknown(guardrailTrace.action);
+  if (action) {
+    guardrailTraceData.action = action;
+  }
+
+  if (
+    "inputAssessments" in guardrailTrace &&
+    isArrayOfObjectWithStringKeys(guardrailTrace["inputAssessments"])
+  ) {
+    guardrailTraceData.inputAssessments = guardrailTrace.inputAssessments;
+  }
+
+  if (
+    "outputAssessments" in guardrailTrace &&
+    isArrayOfObjectWithStringKeys(guardrailTrace.outputAssessments)
+  ) {
+    guardrailTraceData.outputAssessments = guardrailTrace.outputAssessments;
+  }
+
+  return guardrailTraceData;
+}
+
+/**
+ * Determine whether an agent invocation was blocked by any intervening guardrails
+ * @param guardrails Array of guardrail objects to check
+ * @returns True if any guardrail is blocked, false otherwise
+ */
+export function isBlockedGuardrail(guardrails: StringKeyedObject[]): boolean {
+  for (const guardrail of guardrails) {
+    const inputAssessments = Array.isArray(guardrail.inputAssessments)
+      ? guardrail.inputAssessments
+      : [];
+    const outputAssessments = Array.isArray(guardrail.outputAssessments)
+      ? guardrail.outputAssessments
+      : [];
+    const assessments = [...inputAssessments, ...outputAssessments];
+
+    for (const assessment of assessments) {
+      // Check each of the assessment policy types to see if the guardrail is blocked
+      if (isAssessmentBlocked(assessment, "contentPolicy", ["filters"])) {
+        return true;
+      }
+      if (
+        isAssessmentBlocked(assessment, "sensitiveInformationPolicy", [
+          "piiEntities",
+          "regexes",
+        ])
+      ) {
+        return true;
+      }
+      if (isAssessmentBlocked(assessment, "topicPolicy", ["topics"])) {
+        return true;
+      }
+      if (
+        isAssessmentBlocked(assessment, "wordPolicy", [
+          "customWords",
+          "managedWordLists",
+        ])
+      ) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+/**
+ * Parses through guardrail assessment to determine if the action is BLOCKED
+ * @param assessment The assessment object to check
+ * @param policyType The type of policy to check
+ * @param policyFilters Array of filter types to check
+ * @returns True if the assessment is blocked, false otherwise
+ */
+function isAssessmentBlocked(
+  assessment: StringKeyedObject,
+  policyType: string,
+  policyFilters: string[],
+): boolean {
+  const policy =
+    getObjectDataFromUnknown({ data: assessment, key: policyType }) || {};
+
+  const filters: StringKeyedObject[] = [];
+  for (const filterType of policyFilters) {
+    const filterArray = Array.isArray(policy[filterType])
```

Comment:
i don't think this check is accurate to the type, the check here i think should be


```suggestion
    const filterArray = isArrayOfStringKeyedObjects(policy[filterType])
```